src/lib/file-system/file-cache.ts

```diff
@@ -48,4 +47,14 @@ export class FileCache {
 
     return entry;
   }
+
+  updateAngularDiagnostics(sourceFile: ts.SourceFile, diagnostics: ts.Diagnostic[]): void {
+    if (this.has(sourceFile.fileName)) {
```

Comment:
I notice in the other methods that `this.normalizeKey` is used; we should probably do that here too such that deletions align with the keys used here.

Reply:
It’s redundant here,  since both `this.has` and `this.get` will call `normalizeKey` Internally

Reply:
D'oh! Of course :-)